lib/dispatcher.js

```diff
@@ -74,66 +71,63 @@ class DispatcherClient {
     );
   }
 
-  #makeRequest(logContext, url, method, requestBody, cb) {
+  async #makeRequest(logContext, url, method, requestBody, cb) {
+    const requestId = uuid();
     // version *must* be provided
     const urlWithVersion = new URL(url);
     urlWithVersion.searchParams.append('version', this.#version);
     url = urlWithVersion.toString();
-    let body = '';
-    let statusCode = -1;
-    let headers = {};
-    request({
-      url,
-      method,
-      headers: {
-        'Content-Type': 'application/vnd.api+json',
-        Connection: 'Keep-Alive',
-        'Keep-Alive': 'timeout=60, max=10',
-      },
-      body: JSON.stringify(requestBody),
-    })
-      .on('error', (e) => {
+    try {
+      const response = await axiosInstance.request({
+        url,
+        method,
+        data: requestBody ?? JSON.stringify(requestBody),
+        headers: {
+          'Content-Type': 'application/vnd.api+json',
+          Connection: 'Keep-Alive',
+          'Keep-Alive': 'timeout=60, max=10',
+          'Snyk-Request-Id': requestId,
+        },
+      });
+      const statusCode = response.status;
+      const headers = response.headers;
+      const body = response.data;
+      if (statusCode >= 300) {
```

Comment:
we're sure that if we are here it's because the 3 retries failed, and not just the first one while the 2 other keep going in a promise or something?